File: problem/Container_Most_Water/maxArea.c
int min(int a, int b)
{
    return (a < b) ? a : b;
}

int max(int a, int b)
{
    return (a > b) ? a : b;
}

int maxArea(int *height, int heightSize)
{
    int maxAreaValue = 0;
    int left = 0;
    int right = heightSize - 1;

    while (left < right)
    {
        int currentArea = (right - left) * min(height[left], height[right]);
        maxAreaValue = max(maxAreaValue, currentArea);

        if (height[left] < height[right])
        {
            left++;
        }
        else
        {
            right--;
        }
    }

    return maxAreaValue;
}

File: problem/Container_Most_Water/maxArea.py
class Solution:
    def maxArea(self, height):
        max_area = 0
        left, right = 0, len(height) - 1

        while left < right:
            max_area = max(max_area,
                           (right - left) * min(height[left], height[right]))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_area
